iceoryx_binding_c/source/c_subscriber.cpp

```diff
@@ -38,6 +38,13 @@ extern "C" {
 #include "iceoryx_binding_c/subscriber.h"
 }
 
+/// @todo iox-#1221 remove this workaround needed for CycloneDDS due to our change to use the head for storage
```

Comment:
The CycloneDDS workaround now already lasted about one year. Let us see how long this one will last. ;-)

Certainly not pretty but functional.

Resolve me.